src/App.tsx

```diff
@@ -1,26 +1,209 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useState, useEffect } from 'react';
+import cn from 'classnames';
 
-const USER_ID = 0;
+import { UserWarning } from './UserWarning';
+import { USER_ID } from './api/todos';
+import Header from './components/Header';
+import Footer from './components/Footer';
+import TodoList from './components/TodoList';
+import { Status } from './types/Status';
+import * as postService from './api/todos';
+import { Todo } from './types/Todo';
+import { handleError } from './components/Error';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<Status>(Status.All);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [inputDisabled, setInputDisabled] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [shouldFocusInput, setShouldFocusInput] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    postService
+      .getTodos()
+      .then(fetchedTodos => {
+        setTodos(fetchedTodos);
+      })
+      .catch(() => {
+        handleError('Unable to load todos', setErrorMessage);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
+  useEffect(() => {
+    if (shouldFocusInput) {
+      setShouldFocusInput(false);
+    }
+  }, [shouldFocusInput]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const itemsLeft = todos.filter(({ completed }) => !completed).length;
+  const haveCompletedTodos = todos.some(({ completed }) => completed);
+  const allCompleted = todos.every(({ completed }) => completed);
+
+  const filteredTodos = todos.filter(task => {
+    switch (status) {
+      case Status.Active:
+        return !task.completed;
+      case Status.Completed:
+        return task.completed;
+      case Status.All:
+      default:
+        return true;
+    }
+  });
```

Comment:
Create a function that will do it and move this function to utils/helpers file/folder